tests/nix-tests/default.nix

```diff
@@ -8,11 +8,61 @@ let
     inherit yarn2nix;
   };
 
-  call = p: import p {
-    inherit pkgs nixLib yarn2nix;
-  };
+  inherit (import vendor/runTestsuite.nix { inherit pkgs; })
+    runTestsuite
+    it
+    assertEq
+    ;
+
+  # small test package.json
+  my-package-json = pkgs.writeText "package.json" (builtins.toJSON {
+    name = "my-package";
+    version = "1.5.3";
+  });
+
+  # convert a package.json to yarn2nix package template
+  template = package-json: pkgs.runCommandLocal "generate-template" {} ''
+    ${yarn2nix}/bin/yarn2nix --template ${package-json} > $out
+    echo "template for ${package-json} is:" >&2
+    cat $out >&2
+  '';
+
+  # test suite
+  tests = runTestsuite "yarn2nix" [
+    (it "checks the template output"
+      (let tmpl = import (template my-package-json) {} {};
+      in [
+      # TODO: this is a naïve match, might want to create a better test
+      (assertEq "template" tmpl {
+        name = "my-package";
```

Comment:
Note that this should actually be `key = { name = "my-package"; scope = "" };`, but currently `yarn2nix` generates this wrong output.